game.rb

```diff
@@ -0,0 +1,38 @@
+class Game
+
+  require_relative 'score'
+
+  attr_reader :frames
+
+  def initialize
+    @frames = Array.new(10){ [] }
+    @frame_number = 1
+    @roll_number = 1
+    @score = Score.new
+  end
+
+  def roll(pins)
+    if @frame_number < 10
+      if pins < 10 && @roll_number == 1
+        @frames[@frame_number -1] << pins
+        @roll_number = 2
+      elsif pins < 10 && @roll_number == 2
+        @frames[@frame_number -1] << pins
+        @frame_number += 1
+        @roll_number = 1
+      else
+        @frames[@frame_number -1] << pins
+        @frame_number += 1
+      end
+    else
+      @frames[9] << pins
+    end
+  end
+
+  def score

```

Comment:
There's a nice separation of concerns - you've got this adding up the total score - but other parts of the program computing all the complex stuff. A less elegant program would probably have all the score calculated in one method (and correspondingly - any minor tweaks to the program would have knock-on effects through the program). As far as I can tell - you've avoided some of those hazards.